config/confighttp/confighttp_test.go

```diff
@@ -644,7 +759,7 @@ func TestHttpReception(t *testing.T) {
 					return rt, nil
 				}
 			}
-			client, errClient := hcs.ToClient(componenttest.NewNopHost(), component.TelemetrySettings{})
+			client, errClient := hcs.ToClient(componenttest.NewNopHost(), component.TelemetrySettings{Logger: zap.NewNop()})
```

Comment:
Why not using `componenttest.NewNopTelemetrySettings()`?

Reply:
The reason is same as the `tt.TracerProvider = nil`. This tests needs to modify the fields of `http.Transport`. I think I can update the tests to make them more consistent.

Reply:
Do we need to provide a way for those usecases to access the underlying`http.Transport`? Omitting `TracerProvider` or `MetircsProvider` to access `http.Transport` in the `CustomRoundTripper` seems depending on implementation details of `ToClient()`.

Maybe add an option like `WithHTTPTransport(t *Transport)`. But I'm not sure if there are any other use cases besides testing.